### ISSUE TITLE
Tabbing on calendar should force the "on hover" content to display when the using the "date" button style.

### ISSUE THREAD

**Issue by bh3605** [add2cal/add-to-calendar-button#579]
Right now when your button is using the date button style you have this content displayed when the button is hovered over.

![image](https://github.com/add2cal/add-to-calendar-button/assets/3073436/f95bf147-1b2c-4201-859e-966596aa8925)

According to, https://www.w3.org/WAI/WCAG21/Understanding/content-on-hover-or-focus, this content should be displayed when you focus on it as well

**Comment by jekuer** [comment]
I would argue that it is correct that way.
The current hover state is no tooltip as primarily referenced in the article. It is rather an alternative state, specific to pointer events.

Using this also on focus would lead to a different/bad experience for keyboard users.
Idea of the date button is to show the date details. When navigating via keyboard and switching state on focus, users in many cases would have no chance to see this. This would kill the purpose of this button style for them.

The goal of accessibility is imho to provide a good experience for everybody, not mirroring the same journey at all cost.

How do you think about it?

**Comment by bh3605** [comment]
Do you mean the current hover state isn't a tooltip? Just want clarification because your current hover state displays that text over the event details and "current hover state is no tooltip..." sounds off.

I updated this library yesterday and noticed you added an aria label to the button. That's great because now screen readers say "Add to calendar <event name>". What it was doing before is reading all of the content inside the button. That included the date, title, location, and time span. I found it didn't read "Add to Calendar" unless you opened the list, tabbed once, then esc causing the screen reader to read all of the event details again followed by "plus add to calendar collapsed". I mention this because if your current state only reads "Add to calendar <event name>" and you'd expect it to read the event details like date, location, and time when you tab to it then it's not doing that today.

Now I do agree the date style is designed to show all the details (which is exactly why I'm using it) the aria label on the button prevents screen readers from reading all of those event details. Are you suggesting screen readers should be reading those too?

**Comment by jekuer** [comment]
Yes - I wouldn't see it as a regular tooltip, as it covers/replaces the whole element and does not pop up in advance.
So, from the visual side, I would prefer not to show it on focus. This would be the keyboard+noScreenReader case.
When we are talking about the screen reader case: Yes, we should add the event details to the aria label to be clean here.
It would then have the "Add to Calendar" included.

I guess with adding the details to the aria label for the button, we do not need to show the "Add to Calendar" label on focus, right?

**Comment by jodyseger** [comment]
Your suggestion is a solution for screen reader users, but it doesn't cover all the bases for keyboard-only users. Regardless of if this is a tooltip or not, those users are still missing visual context ("Add to Calander") on a focusable element that suggests interaction. In which we aren't able to meet the WCAG 2.1.1 Keyboard (Level A) standard.

I'm thinking the expected interaction for a keyboard-only user would be to show the hover triggered on keyboard focus, to then have the ability to esc from hover focus, for the element to go back to the appointment detail view, while still focused on the element.  

Do you believe this element has that ability? If so, (with the additional aria context on date, title, location, and time) this is a solution for both screen reader users and keyboard-only users and can allow access for a range of disabilities and their assistive technologies.

**Comment by jekuer** [comment]
My point there is: If we prompt the keyboard user with this overlay immediately, we hide the more important information!!

Because it is not "additional content" (aka tooltip), but alternative pointer-adaptive content.
Imagine you start at the top of a page, not yet seeing the button, hit tab and you get to the button already in this second state. It would lose its whole purpose of displaying the date details - from a UX perspective.
(As esc is reserved for closing a potential dropdown, it would also be not intuitive here.)

When hitting the button (space/enter) and the list opens in a dropdown style, we can show the second state. It fits the journey here.

As of the WCAG, this action depends on the user's movement, not the endpoint. That's why imho it is still valid to not show it (as per WCAG definition). Especially as it has no functionality.

**Comment by jekuer** [comment]
Actually, the more I read into the WCAG docs again, the more I am convinced that the current implementation is the best approach. 😅

One more example: If we would consider it a tooltip, it would fail the requirements for a good tooltip, as it hides the content 😉. It is fine in the pointer case, but not the keyboard case.

**Comment by bh3605** [comment]
Regardless of whether you decided to implement this or not, the button's aria-label is preventing screen readers from reading the event details so if you don't consider the focused interaction a bug, unintentionally preventing screen readers from reading all of the event content still is. FWIW, if you fix this bug, make sure screen readers are able to pronounce the month. Before I updated, I had to assign a role for the screen reader to say "April" and not "a-p-r". I'm sure you guys have an enum or something somewhere mapping "April" to "Apr" so to set it up so the screen reader says "April" __should be easy__™.

**Comment by jekuer** [comment]
Alright.
So, for now, we pass on showing the slogan on focus. However, we optimized the Aria Label to show full information, including the slogan you get on hover. This is somehow a compromise and should cover all cases best for now.

We did not optimize the wording as this is a bigger feature request, not a bug.
We are also struggeling with the common issues screen readers have with the shadow DOM.
Will optimize this in a later major version as this is been done best with a complete refactoring.

Check out v2.6.10!
